platforms/core-configuration/model-core/src/integTest/groovy/org/gradle/api/provider/PropertyIntegrationTest.groovy

```diff
@@ -191,10 +192,16 @@ task thing(type: SomeTask) {
 
         then:
         failure.assertHasDescription("Execution failed for task ':thing'.")
-        failure.assertHasCause("""Cannot query the value of task ':thing' property 'prop' because it has no value available.
+        def baseMessage = "Cannot query the value of task ':thing' property 'prop' because it has no value available."
```

Comment:
Made the test conditional on the executer not being `configCache` with a proper explanation. Since extensions do not exist at execution time (with CC enabled), lack of provenance info involving extensions is not an unreasonable outcome.